core/src/main/java/com/coderanch/blackjack/Card.java

```diff
@@ -12,31 +12,66 @@
  */
 final class Card {
   
+  enum Rank { ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING };
+  enum Suit { SPADES, HEARTS, CLUBS, DIAMONDS };
+  
+  private final Rank rank;
+  private final Suit suit;
+  
   /**
-   * The rank of a {@link Card}.
-   */
-  enum Rank {}
-
-  /**
-   * The suit of a {@link Card}.
+  * Constructor.
+  *
+  * @param rank (required) Rank value. Must have value of enum Rank. 
+  * @param suit (required) Suit value. Must have value of enum Suit.
+  */
+  
+  public Card(Rank rank, Suit suit) {
+    this.rank = getConstRank(rank);
+    this.suit = getConstSuit(suit);
+  }
+  
+  /*
+  * Oracle states that private methods should not have a JavaDoc, 
+  * also that assertions should not be used for runtime exceptions.
+  * I have implemented these getters as I;ve been told logic should not be placed
+  * in the constructor itself. I'm not sure if it is okay to do it this way? 
+  * Any feedback would be geat. Also, suit not being of Suit is an error (cannot find symbol).
+  * How is this supposed to be checked for without surrounding constructor in try catch?
```

Comment:
It's good practice to write Javadoc for private methods and constructors anyway, because people other than yourself may want to call them and they want to know what those operations are supposed to do.

You can put logic in the constructor of that logic is necessary to initialize the object. When people say not to put logic in the constructor, they mean additional operations besides parameter checking and field initialization.

Please don't put questions about the code in the code. Push the code and then annotate it in GitHub, like I'm doing right now.